frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -583,6 +587,44 @@ pub mod pallet {
 		/// this value, based on [`WeightInfo::submit_unsigned`].
 		type MinerMaxWeight: Get<Weight>;
 
+		/// Maximum number of signed submissions that can be queued.
+		///
+		/// It is best to avoid adjusting this during an election, as it impacts downstream data
+		/// structures. In particular, `SignedSubmissionIndices<T>` is bounded on this value. If you
+		/// update this value during an election, you _must_ ensure that
+		/// `SignedSubmissionIndices.len()` is less than or equal to the new value. Otherwise,
+		/// attempts to submit new solutions may cause a runtime panic.
+		#[pallet::constant]
+		type SignedMaxSubmissions: Get<u32>;
```

Comment:
As you have also noticed recently, it is a big PITA to go through a full runtime upgrade cycle in order to make any changes. I am becoming more a fan of making as many of configs as possible, a storage item, and a a root controlled call to set them/each.

Note that this should only be possible for configs that are transient. For example, if deposits are recorded on-chain, then we can change the deposit on the fly. But If it weren't, this would have not been valid. 

I think `SignedMaxSubmissions`, `SignedRewardBase`, and the 3 deposits could be storage items. WDYT?

Reply:
Yeah, I can get behind that. I'm going to ask around in the Matrix W3F channel to get opinions for the values.

Reply:
We have to be smart about when to use such things. There is a non-zero overhead to making things storage items, and this sees to me to be a configuration which will be very infrequently updated. But just something to consider, not saying one way or another.

Reply:
I don't have strong feelings one way or the other about whether they should be storage items or consts. There are reasonable arguments on both sides.

This PR seems unlikely to make it into v0.9.5, which means that we have some time to make the decision. If we do decide to make them into storage items, that's a pretty easy change--easier, at least, than the reverse.

Therefore, extracted this into https://github.com/paritytech/substrate/issues/9185 so we can talk about it there. In the meantime, I'd prefer not to block this PR on a resolution of that decision.

Reply:
> This PR seems unlikely to make it into v0.9.5 

9.5 was finalized 4 days ago actually :p https://github.com/paritytech/polkadot/releases/tag/v0.9.5.

> Therefore, extracted this into #9185 so we can talk about it there. In the meantime, I'd prefer not to block this PR on a resolution of that decision.

agreed 👍